calico_containers/pycalico/datastore_errors.py

```diff
@@ -51,4 +51,10 @@ class MultipleEndpointsMatch(Exception):
     """
     More than one endpoint was found for the specified criteria.
     """
+    pass
+
+class InvalidPoolError(Exception):
```

Comment:
Can you rename to something more specific - this makes it sound like the pool was invalid, but that's not the case - you only throw this exception when there is no matching Pool.